pkg/obfuscate/shell.go

```diff
@@ -0,0 +1,607 @@
+				if !stringInSlice(token.Val, allowedEnvVars) {
+					tokens[index].Val = result[1] + "=?"
+					obfuscatedArgsIndices = append(obfuscatedArgsIndices, strconv.Itoa(index)+"-"+strconv.Itoa(len(result[1])+1))
+				}
+			} else {
+				// If not formatted like an env variable, likely the binary
+				if stringInSlice(token.Val, processDenyList) {
+					// Remove every parameter until the end of the command
+					if isExecCmd {
+						// Remove all tokens until the end (the whole command)
+						for index++; index < len(tokens); index++ {
+							tokens[index].Val = "?"
+							obfuscatedArgsIndices = append(obfuscatedArgsIndices, strconv.Itoa(index))
+						}
+					} else {
+						for index++; index < len(tokens) && tokens[index].Kind == Field; index++ {
+							tokens[index].Val = "?"
+							obfuscatedArgsIndices = append(obfuscatedArgsIndices, strconv.Itoa(index))
+						}
+					}
+				}
+
+				foundBinary = true
+			}
+		} else { // Alright, we’re in the parameters then
+			// Are we in the case of --pass=xxx or --pass xxx
+			if equalIndex := strings.Index(token.Val, "="); equalIndex == -1 {
+
+				// if --pass xxx, check is --pass is allowed and that we have a xxx
+				if regexParamDeny.MatchString(token.Val) && index < len(tokens)-1 {
+					(tokens)[index+1].Val = "?"
+					obfuscatedArgsIndices = append(obfuscatedArgsIndices, strconv.Itoa(index+1))
+				}
+
+				skipNext = true
+			} else {
+				// split at the first equal sign
+				param := token.Val[:equalIndex]
+
+				if regexParamDeny.MatchString(param) {
+					tokens[index].Val = param + "=?"
+					obfuscatedArgsIndices = append(obfuscatedArgsIndices, strconv.Itoa(index)+"-"+strconv.Itoa(len(param)+1))
+				}
+			}
+		}
+	}
+
+	return obfuscatedArgsIndices
+}
+
+// findShellCommand search for the shell command in the given shell struct of the cmd.exec command
+func findShellCommand(commandArgs []string) int {
+	// Search for the -c parameter in the command arguments to detect if there is a shell execution
+	var cArgDetected bool
+	var argMayWaitForArg bool
+	for index, arg := range commandArgs {
+		// The argument -c can have other arguments inside the same string
+		// If it's starting with a dash, it's a parameter, then check if the letter 'c' is inside
+		if len(arg) > 0 && arg[0] == '-' {
+			// That's a parameter
+			argMayWaitForArg = true
+
+			// Ignore double dashed parameters (they can have a value)
+			if len(arg) > 1 && arg[1] != '-' && strings.Contains(arg, "c") {
+				cArgDetected = true
+			}
+		} else {
+			// That's a command
+			if cArgDetected {
+				return index + 1 // +1 to skip the -c parameter
+			}
+
+			if argMayWaitForArg {
+				argMayWaitForArg = false
+				continue
+			}
+
+			// Not an injected shell command via -c
+			return -1
+		}
+	}
+
+	return -1
+}
+
+// stringInSlice checks if the given string is present in the given slice
+func stringInSlice(str string, list []string) bool {
```

Comment:
can we just use `slices.Contains`? (or is that worse performance due to generics?)